file_formats/cmt_format.ml

```diff
@@ -60,21 +79,112 @@ type cmt_infos = {
   cmt_imports : (string * Digest.t option) list;
   cmt_interface_digest : Digest.t option;
   cmt_use_summaries : bool;
-  cmt_uid_to_loc : Location.t Shape.Uid.Tbl.t;
+  cmt_uid_to_decl : item_declaration Shape.Uid.Tbl.t;
   cmt_impl_shape : Shape.t option; (* None for mli *)
+  cmt_usages_index : (index_item * Longident.t Location.loc) list
```

Comment:
I just had a discussion with Ulysse about the representation of `cmt_usages_index` and that made me realize something I had missed: that we store a `Longident.t` and not a `Path.t`, which is somewhat surprising.

We discussed it a bit, changing to `Path.t` would be a bit annoying but also isn't obviously the right call. At least the lid represents what's in the source.
In the end, my intuition is that here only the location is meaningful. And perhaps that's the only thing that should be stored?